assets/js/services/hooks/use-site-options/index.js

```diff
@@ -0,0 +1,91 @@
+import { useEffect, useState } from 'react';
+import apiFetch from '@wordpress/api-fetch';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+const useSiteOptions = () => {
+  const [loading, setLoading] = useState(true);
+  const [notices, setNotices] = useState([]);
+  const [saving, setSaving] = useState(false);
+  const [settings, setSettings] = useState({});
+
+  // Setup for Gutenberg's notices system.
+  const {
+    createErrorNotice,
+    createSuccessNotice,
+    removeNotice,
+  } = useDispatch('core/notices');
+  const noticeOptions = {
+    type: 'snackbar',
+    isDismissable: true,
+  };
+
+  /**
+   * Helper for creating an error notice and adding it to the stack.
+   * @param {string} message - The message to display to the user.
+   */
+  const error = async (message) => {
+    const { notice: { id } = {} } = await createErrorNotice(message, noticeOptions);
+    setNotices([...notices, id]);
+  };
+
+  /**
+   * Helper for creating a success notice and adding it to the stack.
+   * @param {string} message - The message to display to the user.
+   */
+  const success = async (message) => {
+    const { notice: { id } = {} } = await createSuccessNotice(message, noticeOptions);
+    setNotices([...notices, id]);
+  };
+
+  // Load settings on initial mount.
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await apiFetch({
+          path: '/wp/v2/settings',
+        });
+        setSettings(response || {});
+      } catch ({ message }) {
+        await error(message);
+      } finally {
+        setLoading(false);
+      }
+    })();
+  }, []);
```

Comment:
Need an eslint-disable here for the rule about the missing dependency array (which in this case is intentional)